### PR TITLE
Invert bootsplash colors + fade splash screen on web

### PR DESCRIPTION
**Pull request by zoontek**
### Details

Following [this discussion](https://github.com/Expensify/App/pull/16863#issuecomment-1494154090) with @shawnborton, he thought it could be great to invert splash screen colors, in orders to see the app icon "grow" into the splash screen.

This PR provides that + as a bonus, it adds fading on the web splash screen (for consistency with native) - I can revert the commit if this is not wanted.

### Fixed Issues

https://github.com/Expensify/App/pull/16863#issuecomment-1494154090

### Tests

- ⚠️ Checkout https://github.com/Expensify/App/pull/16863 branch, then merge this one in it.
- Starts the app on Android, iOS or web
- Verify that the splash screen colors are inverted

### QA Steps

Web
1. Open the App
2. Verify that the splash screen had a dark green background, light green logo, and fades away.

iOS and Android
1. Open the App
2. Verify that the splash screen had a light green background, dark green logo, and fades away.

No QA for other platforms.
### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/1902323/229892172-1ba4202c-c4ef-4b89-b3fa-4cea15a5392d.mov

</details>

<details>
<summary>iOS</summary>

Screenshot 2023-04-08 at 11 26 49

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/1902323/229883697-76048c20-ea8d-4b84-9aef-89ed2982809c.mp4

</details>

### CODE REVIEW

**Comment by shawnborton** [comment]
Is it possible to only do this on native devices? For web, I feel like the green is pretty distracting. But for mobile, it's nice because it looks like the app icon morphs into the splash screen.

**Comment by zoontek** [comment]
@shawnborton I rolled back the change, but kept the fade.

https://user-images.githubusercontent.com/1902323/229892172-1ba4202c-c4ef-4b89-b3fa-4cea15a5392d.mov

**Comment by shawnborton** [comment]
Nice, I think that feels pretty good to me

**Comment by neil-marcellini** [comment]
Tagging in a C+ for some extra review help https://expensify.slack.com/archives/C02NK2DQWUX/p1680713244754089?thread_ts=1680713226.483109&cid=C02NK2DQWUX

**Review by neil-marcellini — COMMENTED** [review]
@zoontek it looks like you included the changes from https://github.com/Expensify/App/pull/16863 here correct? I think it would be much more simple to review if you remove those changes and instruct us reviewers to check out the branch from 16863, and then merge these changes into it for testing.

**Comment by zoontek** [comment]
@neil-marcellini @0xmiroslav Yes. I can dissociate both PR if you prefer, remove the changes from https://github.com/Expensify/App/pull/16863 from this one

**Comment by zoontek** [comment]
@neil-marcellini @0xmiroslav Done! I updated the instructions.

**Comment by zoontek** [comment]
@0xmiroslav You cannot merge this one before https://github.com/Expensify/App/pull/16863 since the image format change for Android 12: they must include an additional padding:

Screenshot 2023-04-06 at 11 31 38

If you want to merge this one before, I will have to update this PR to re-generate assets in this PR to not include the additional padding, wait for it to be merged then update the other PR. It's easier to keep this one on hold. The other PR is not that huge, I think the review will probably be done soon.

**Comment by zoontek** [comment]
@0xmiroslav @neil-marcellini #16863 has been merged, I rebased this one.

**Review by neil-marcellini — DISMISSED** [review]
Much easier to review now, thank you! Looks good and tests well.

**Comment by neil-marcellini** [comment]
@0xmiroslav all you!

**Comment by zoontek** [comment]
@0xmiroslav Yes, here: https://github.com/Expensify/App/pull/16932#issuecomment-1496433065

I opened the PR with light splash screen on web, but had to rollback this part.

File: web/splash/splash.js [hunk 1]
```diff
@@ -26,8 +26,14 @@ document.addEventListener('DOMContentLoaded', () => {
         passedMiliseconds += 250;
         isRootMounted = root.children.length > 0;
         if (passedMiliseconds >= minMilisecondsToWait && isRootMounted && areFontsReady) {
-            splash.parentNode.removeChild(splash);
             clearInterval(intervalId);
+
+            splash.style.opacity = 0;
+            splash.style.pointerEvents = 'none';
```
[the remark this answers is not in the document]
Comment (zoontek) [reply web/splash/splash.js:32]: pointerEvents: none will allow the user to click though the splash screen when it starts to fade.

Sure, 250ms is short, but we never know 🙂

[the remark this answers is not in the document]
Comment (zoontek) [reply web/splash/splash.js:32]: (I can also remove this line if you don't want this behavior)

**Comment by zoontek** [comment]
@0xmiroslav Thanks for the review!

1. This is a cache issue. Since Android 12, the app is started on system thread for snappier boot. This explain the double splash screen that you had before. To "fix" this: uninstall the app, restart your phone (as the first assets is cached by the system), clean your gradle cache, reinstall the app.

2. It always as been there, and has not been added in this PR, or the other. It's just that as the splash screen is a bit quicker to hide (lighter than an Activity, no additional delay), it's now visible from time to time. You should remove it 🙂

3. This has been Android choice to show the splash screen with fully transparent system bars on system thread. What you saw here is: 1. system splash screen is shown, transparent bars 2. the app boot and as I just set a statusBar color, the navigation bar change color (take your app theme value, which is no value). I didn't want to make a choice for that since your app does not use transparent navigation bar, but we can set its color too if needed. Before, as you had two different splash (the system one and your custom), you didn't notice that it occurs because, well… the whole splash changed, not just the bars 😄

4. Does this happen only during the splash screen shown / bundle load?

File: web/splash/splash.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (zoontek) [reply web/splash/splash.js:32]: Not if the splash screen doesn't start fading. But I will remove it, for 250ms it's like micro-optimization

**Comment by zoontek** [comment]
@neil-marcellini Restarted your phone between the two?

File: web/splash/splash.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (zoontek) [reply web/splash/splash.js]: No.

opacity: 1 -> the splash is visible, no click go though it
opacity between 1 to 0 (during the fade) -> click though allowed
opacity 0 -> the splash is hidden

But I removed it, as it wasn't asked.

**Comment by zoontek** [comment]
@0xmiroslav Could you try uninstalling the dev app, restart your phone and reinstall the same app (the dev one?)

I updated the navigationBar color to match the rest. It swithes from dark content to light content once the app booted (the system used dark content but your app uses light, you can tweak that as you want in themes files).

https://user-images.githubusercontent.com/1902323/230791915-e1afd7e1-f6f4-4401-867c-385cfba168b5.mp4

I also fixed 4. Android documentation mention that both application theme or main activity theme are fine, which is the mistake here (see [step 3 here](https://developer.android.com/develop/ui/views/launch/splash-screen/migrate#migrate_your_splash_screen_implementation))

**Comment by zoontek** [comment]
@neil-marcellini @0xmiroslav I pushed the missing `AppTheme` on application. Sorry, I forgot to put it back!

**Review by 0xmiroslav — APPROVED** [review]
Looks good and tests well but only one issue already reported.
I confirmed that splash logo changing issue still happens on fresh android physical device. I tested on Samsung Galaxy S10. So this is not related to cache.
But this was introduced in another PR so not a blocker for this PR. If needed, we can handle that in a new separate issue.
@neil-marcellini all yours

https://user-images.githubusercontent.com/97473779/231463823-2f87053f-e4b8-4976-9de1-7fac4eed71d2.mp4

**Review by neil-marcellini — APPROVED** [review]
I think this is good to go. Thanks.

**Comment by neil-marcellini** [comment]
@zoontek I also added some QA steps for you. Merging!

**Comment by neil-marcellini** [comment]
Sure thing, here's the payment issue.

**Pull request merged**